addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRuleTest.java

```diff
@@ -0,0 +1,328 @@
+import org.junit.jupiter.params.provider.ValueSource;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link SpringActuatorScanRule}. */
+class SpringActuatorScanRuleUnitTest extends ActiveScannerTest<SpringActuatorScanRule> {
+
+    private static final String ACTUATOR_CONTENT_TYPE =
+            "Content-Type: application/vnd.spring-boot.actuator.v2+json;charset=UTF-8";
+    private static final String RELEVANT_BODY = "{\"status\":\"UP\"}";
+    private static final String IRRELEVANT_BODY = "<body><h1>Howdy></h1></body>";
+    private static final String REQ_PATH = "actuator/health";
+
+    @Override
+    protected SpringActuatorScanRule createScanner() {
+        return new SpringActuatorScanRule();
+    }
+
+    @Override
+    protected void setUpMessages() {
+        mockMessages(new ExtensionAscanRulesAlpha());
+    }
+
+    @Test
+    void shouldTargetJavaTech() {
+        // Given
+        TechSet techSet = techSet(Tech.JAVA);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotTargetNonJavaTechs() {
+        // Given
+        TechSet techSet = techSetWithout(Tech.JAVA);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(false)));
+    }
+
+    @Test
+    void shouldBeInfoGatherAlert() {
+        // Given
+        Integer category = Category.INFO_GATHER;
+        // When
+        Integer ruleCategory = rule.getCategory();
+        // Then
+        assertEquals(ruleCategory, category);
+    }
+
+    @Test
+    void shouldSkipScanIfStopped() throws Exception {
+        // Given
+        rule.init(getHttpMessage("/"), parent);
+        parent.stop();
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, is(empty()));
+    }
+
+    @Test
+    void shouldScanMessageWithoutPath() throws HttpMalformedHeaderException {
+        // Given
+        String path = "";
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, RELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = getHttpMessage(path);
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then = no exception
```

Comment:
If no exception is really what you want to check here... though it should probably be that no messages are sent.

```java
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
...
        // When / Then
        assertDoesNotThrow(() ->rule.scan());
```

Reply:
Still ?

Reply:
Ahh, didn't see this one. (got buried in comment folds)